### DISCUSSION TITLE
Pull and place files after choosing ISO

### DISCUSSION THREAD

**Discussion by ospolot** [ventoy/Ventoy#1321] [Ideas]
Hello @Ventoy,

recently I ran into a problem of not being able to pull Files from the point after choosing the ISO.

Currently, I'm trying to auto-install Ubuntu 20.04 with the cidata Partition. I'm getting the user-data file from an HTTP site with a curl.

My goal is to be able to pull the file after choosing the ISO in the ventoy GUI and placing it into the third cidata Partition.

So the auto-install could go on with a dynamic user-data.

Another solution would be to have the chance to execute the script after the Point of choosing.

Already watched the Ventoy Boot Conf Replace Plugin, but you can only refer to local Paths within the partition.

Maybe you also had such an idea. It would be very pleasant of you to give this a thought.

Looking forward to hearing from you.

Greetings

Ospolot

**Comment by ventoy** [comment]
Do you mean that you can not mount the partition where the ISO file locate after you select the ISO and boot?

**Reply by ospolot** [reply]
I have no problem with booting the system. I just want an opportunity to pull a file via wget or curl and place it into the third partition after I selected the ISO. It would have the same time slot as the Boot Conf Replacement Plugin. Like an injection before actually booting the ISO.

**Reply by ventoy** [reply]
You can use auto install plugin and run wget command inside the auto install configuration (e.g. preseed.cfg)
There are some mechanisms for user to run shell command with auto install script.

**Reply by ospolot** [reply]
Unfortunately preseed only works for <20.x. 
Since 20.x Ubuntu-Server uses cloud-init (user-data), not preseed.cfg. 
The auto-install uses the correct order, but it can't modify files or execute scripts.

**Reply by maxdwit** [reply]
Boot -> ventoy -> wget (changes) to files on cidata -> start Ubuntu-Server ISO 
---------------------^^^  at this point he wants to modify user-data to replace some information

With Ubuntu 22.04 coming up, that would also be helpful there. As I understand they will also switch to the new way of installing. Need to verify that.
